main.swift

```diff
@@ -186,16 +200,56 @@ func performCommand(description: String, command: () throws -> Void) rethrows {
     print("✅  Done")
 }
 
+//: This *could* just be a dictionary, but a struct felt cleaner. Also, we get type safety(and autocompletion) for free!!
+struct TemplateInformation {
+    var destination: String? = nil
+    var projectName: String? = nil
+    var authorName: String? = nil
+    var authorEmail: String? = nil
+    var githubURL: String? = nil
+    var organizationName: String? = nil
+}
+
+func processArguments() -> TemplateInformation {
+    var templateInfo = TemplateInformation()
+    
+    let args = CommandLine.arguments
+    
+    for (index,argument) in args.enumerated() {
+        //: Why lowercased() ? In case the user accidentally mistypes something.
+        switch argument.lowercased() {
+        case "--destination".lowercased() :
```

Comment:
This is for a situation where a user may type in `--Destination` instead of `--destination` or something. Just wanted it to be somewhat fault tolerant.